fetools-app/src/components/Heart.jsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from "react";
+
+import HeartIcon from "./HeartIcon";
+
+export default function Heart({addClass, pageName, elementId}){
+
+    checkForLocalStorage(pageName)
+
+    const [heartedItem, setHeartedItem] = useState(false)
+
+    return (
+        <>
+          <HeartIcon onClick={(evt)=>handleClick(evt)} isHearted={updateHeart()} addClass={addClass}/>
+        </>
+    )
+
+    function handleState(){
+        setHeartedItem(!heartedItem)
+    }
+
+    function handleClick(evt){
+        const currentElement = evt.target.closest(`#${elementId}`) //get element with id
+
+        const favoriteItemsJSON = localStorage.getItem(`${pageName}-favorites`)
+        const favoriteItems = JSON.parse(favoriteItemsJSON)
+
+        if(!favoriteItems.includes(elementId)){
+            favoriteItems.push(elementId)
```

Comment:
This component is great. We do need to allow it to be a bit more flexible, so instead of accepting an elementId, lets allow it to accept favoriteContent.

For the character finder, it will simply be a character. For the color picker, it will be a hex color code. For the font visualizer, it will be an object of font settings.